### PR TITLE
Preserve YOLOE one-to-one classifiers during linear probing

### PR DESCRIPTION
**Pull request by Y-T-G**
In `YOLOEPETrainer.get_model`, the three `one2one_cv3` assignments copy from `cv3` instead of `one2one_cv3`:

```python
model.model[-1].one2one_cv3[0][2] = deepcopy(model.model[-1].cv3[0][2]).requires_grad_(True)
```

The two branches have their own `cv3` feature layers, so pairing the one2one features with the one2many final projection destroys the head. For end-to-end models like YOLO26, inference and validation run the one2one branch, so linear probing starts from near-zero mAP instead of the pretrained level.

The `deepcopy` is only there to drop the `requires_grad_(False)` set by `_fuse_tp`, so copying each branch its own conv keeps the pretrained weights.

Validation on COCO128 with `yoloe-26s-seg.pt`, `YOLOEPETrainer`, 1 epoch:

| one2one_cv3 source | mAP50 | mAP50-95 |
| --- | --- | --- |
| `cv3` (before) | 0.018 | 0.012 |
| `one2one_cv3` (after) | 0.546 | 0.403 |

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes YOLOEPETrainer initialization so the one2one classifier retains its own weights instead of being overwritten by one2many classifier weights.

### 📊 Key Changes
- Updates `one2one_cv3` layers to deep-copy their existing classifier modules.
- Preserves independent one2one and one2many classifier weights during model setup.
- Keeps the modified classifier layers trainable with `requires_grad_(True)`.

### 🎯 Purpose & Impact
- Prevents unintended weight sharing or overwriting in YOLOE pose-free training.
- Improves correctness and stability for models using separate one2one and one2many detection heads.
- Limits the fix to the affected initialization logic without changing other training behavior. 🌟

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated welcome message, and an engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The detection-side correction is sound: it preserves the independently fused one-to-one classifier weights while making those projections trainable. However, the equivalent segmentation linear-probing path remains unchanged, so the stated YOLOE-26 segmentation regression is not fully fixed.

💬 Posted 1 inline comment

File: ?
[the remark this answers is not in the document]
Comment (Y-T-G) [reply ?:1]: Fixed. Applied the same correction in `YOLOEPESegTrainer.get_model` (`train_seg.py:105-107`).

Validation on COCO128-seg with `yoloe-26s-seg.pt`, `YOLOEPESegTrainer`, 1 epoch: mAP50-95 **0.403** box, **0.313** mask.

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean PR. The updated assignments preserve each one-to-one classifier projection after text-prompt fusion while restoring it as trainable, matching the separate end-to-end branch structure in both detection and segmentation trainers.
